sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
I know `resolveAndBind` will resolve it. But looks at the code snippet above:

    val deserializer = encoderFor[T].deserializer
    val dataType = deserializer.dataType
    val nullable = if (deserializer.childrenResolved) {  // or deserializer.resolved

I don't see you have a chance to call `resolveAndBind` here...

`encoderFor[T]` will call `assertUnresolved` on the encoder to return, it is implemented like:

    def assertUnresolved(): Unit = {
      (deserializer +:  serializer).foreach(_.foreach {
        case a: AttributeReference if a.name != "loopVar" =>
          sys.error(s"Unresolved encoder expected, but $a was found.")
        case _ =>
      })
    }

It makes sure that both `deserializer` and `serializer` are unresolved.